File: frontend/src/types/player.ts
export interface Player {
  id: string;
  player_code: string;
  first_name: string;
  last_name: string;
  full_name: string;
  display_name: string;
  team_id: number | null;
  team_name: string | null;
  sport_id: number | null;
  sport_name: string | null;
  graduation_year: number | null;
  date_of_birth: string | null;
  is_pitcher: boolean;
  is_position_player: boolean;
  bats: 'R' | 'L' | 'S' | null;
  throws: 'R' | 'L' | null;
  height_inches: number | null;
  weight_lbs: number | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface PlayerListItem {
  id: string;
  player_code: string;
  full_name: string;
  team_name: string | null;
  sport_id: number | null;
  sport_name: string | null;
  is_pitcher: boolean;
  is_position_player: boolean;
  is_active: boolean;
  graduation_year: number | null;
}

export interface PlayerCreate {
  first_name: string;
  last_name: string;
  team_id?: number | null;
  sport_id?: number | null;
  graduation_year?: number | null;
  date_of_birth?: string | null;
  is_pitcher?: boolean;
  is_position_player?: boolean;
  bats?: 'R' | 'L' | 'S' | null;
  throws?: 'R' | 'L' | null;
  height_inches?: number | null;
  weight_lbs?: number | null;
}

export interface PlayerUpdate extends Partial<PlayerCreate> {
  is_active?: boolean;
}

export interface PlayerWithAssessments extends Player {
  assessment_count: number;
  latest_assessment_date: string | null;
  assessment_types: string[];
}

File: backend/app/models/player.py
from sqlalchemy import Column, String, Integer, Boolean, Date, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime
import uuid
from app.db.base import Base


class Player(Base):
    __tablename__ = "players"
    __table_args__ = {"schema": "organization"}

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    player_code = Column(String(20), unique=True, nullable=False, index=True)
    first_name = Column(String(100), nullable=False)
    last_name = Column(String(100), nullable=False)
    team_id = Column(Integer, ForeignKey("organization.teams.id"), nullable=True)
    graduation_year = Column(Integer, nullable=True)
    date_of_birth = Column(Date, nullable=True)

    # Player type flags
    is_pitcher = Column(Boolean, default=False)
    is_position_player = Column(Boolean, default=True)

    # Physical attributes
    bats = Column(String(1), nullable=True)  # R, L, S
    throws = Column(String(1), nullable=True)  # R, L
    height_inches = Column(Integer, nullable=True)
    weight_lbs = Column(Integer, nullable=True)

    # Status
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    team = relationship("Team", back_populates="players")
    assessment_sessions = relationship(
        "AssessmentSession", back_populates="player", cascade="all, delete-orphan"
    )

    @property
    def full_name(self) -> str:
        return f"{self.first_name} {self.last_name}"

    @property
    def display_name(self) -> str:
        team_name = self.team.name if self.team else "No Team"
        return f"{self.full_name} (Team: {team_name})"
